### DISCUSSION TITLE
[Question]: How to disable conceal highlight?

### DISCUSSION THREAD

**Discussion by ayoubelmhamdi** [nvim-treesitter/nvim-treesitter#6208] [Q&A]
I want to use conceal, but the highlighting using treesitter comes with unwanted conceal highlight. 
```lisp
(fenced_code_block
  (fenced_code_block_delimiter) @markup.raw.delimiter
  (#set! conceal ""))

(fenced_code_block
  (info_string
    (language) @conceal
    (#set! conceal "")))
```
I don't want to remove it from the `repo`, because it is hard to update the repo, but is there any way to override this capture query to make the block raw text look like the original?

I searched in GitHub issues for the same problem, but I didn't find any related issue, maybe there is already an issue like this.

**Comment by clason** [comment]
You have to copy the queries (with these patterns removed) to your own config dir; if you haven't messed with things too much, they will then have precedence over both the bundled (with Neovim) and the installed (with nvim-treesitter) queries.

There is no other way. (And be advised that this will break LSP hover and type signature.)

**Comment by ayoubelmhamdi** [comment]
i use `lazy.nvim`
```lua
return {
    'nvim-treesitter/nvim-treesitter',
    event = 'VeryLazy',
    dependencies = {
        { 'lewis6991/spellsitter.nvim' },
        { 'David-Kunz/treesitter-unit' },
        { 'nvim-treesitter/nvim-treesitter-textobjects' },
        { 'JoosepAlviste/nvim-ts-context-commentstring' },
        { 'p00f/nvim-ts-rainbow' },
    },
    build = ':TSUpdate',
    config = function()
        require('nvim-treesitter.configs').setup({
            ensure_installed = { 'markdown', 'markdown_inline', 'javascript', 'typescript', 'c', 'python', 'lua', 'rust' },
            sync_install = false,
            auto_install = true,
            highlight = {
                enable = true,
                additional_vim_regex_highlighting = true,
            },
        })

        require('nvim-treesitter.install').prefer_git = true
        local parser_config = require('nvim-treesitter.parsers').get_parser_configs()
        parser_config.typst = {
            install_info = {
                url = '/data/local_tmp/github_treesitters/tree-sitter-typst',
                files = { 'src/parser.c', 'src/scanner.c' },
            },
            filetype = 'typst',
        }
    end,
}
```


I have this text file called md.md with set `conceallevel = 0`
<img width=120 src="https://github.com/nvim-treesitter/nvim-treesitter/assets/34358145/6caea94d-e72a-4afc-93e2-c95eb7a298ce">



I tried to override the conceal character by using this `scm` file in `after/queries/markdown/highlights.scm` path:

```lisp
(fenced_code_block
  (fenced_code_block_delimiter) @markup.raw.delimiter
  (#set! conceal "+"))

(fenced_code_block
  (info_string
    (language) @conceal
    (#set! conceal "+")))
```
but it didn’t override yet. 
<img width=120 src="https://github.com/nvim-treesitter/nvim-treesitter/assets/34358145/c3b602dc-5920-45fc-9468-ae70b49a2998">

**Reply by clason** [reply]
Do not place the file in `after`. If you want to override a single pattern, you must put `; extends` at the top of the file (see the Neovim documentation). I'm not sure you _can_ override conceals this way, though; you certainly cannot _remove_ conceal. This is why I told you to copy the _full_ query (without `; extends`) and make modifications there.

**Reply by ayoubelmhamdi** [reply]
When we place any query in `plugin/queries/markdown/highlights.scm` file, nothing happens. but When we place a query in `after/queries/markdown/highlights.scm`, the query executes (if not exist), and the `;extends` directive doesn't help to override at all.

**Reply by clason** [reply]
Not under `plugin`, either. Just `.config/nvim/queries`.

**Reply by ayoubelmhamdi** [reply]
Thank you, @clason, it works fine for me now.

To summarize how we did it:
When we use the `lazy.nvim` plugin manager, and we want to override (not extend) any highlight (or folds, highlights, indents, injections) that was created by the `nvim-treesitter` neovim plugin at `~/.local/share/nvim/lazy/nvim-treesitter/queries/markdown/highlights.scm`, we should not use the `;extends` keyword, but we should copy the entire file into `~/.config/nvim/queries/markdown/highlights.scm` to prevent some future problems, than we can modify manually the copied file to what we want.
